### DISCUSSION TITLE
Implementation of MA ids to Human-GEM

### DISCUSSION THREAD

**Discussion by haowang-bioinfo** [SysBioChalmers/Human-GEM#260] [Ideas]
As planned and discussed in #174, MA ids for both reactions and metabolites are ready to be used.

Now seem to be a good time point for their implementation. But still would be good know if there are any potential issues or preparations associated with this transition. @all comments/questions are welcome.

**Comment by mihai-sysbio** [comment]
Here are some thoughts:
- will there be any versioning, eg what happens when the ID stays the same, but some details change (say associated annotation)?
- how will consistency be ensured between versions, eg an ID is used, then removed, and then a new element is added with the same ID?

**Reply by haowang-bioinfo** [reply]
@mihai-sysbio good comments. Here are my thoughts:

> will there be any versioning, eg what happens when the ID stays the same, but some details change (say associated annotation)?

The versioning and change tracking to MA ids (and other model components) can be achieved by the built-in feature of `Git` via GH. And it's the responsibility of reviewer to make sure the details is reasonable.

> how will consistency be ensured between versions, eg an ID is used, then removed, and then a new element is added with the same ID?

As changes can be always evaluated by reviewers and tracked on GH, the consistency between model components and files (`yaml` <-> `tsv`) is ensured by existing code (e.g. `sanityCheck`) and new ones whenever necessary.

**Reply by mihai-sysbio** [reply]
That's not the interpretation of versioning I had in mind, maybe I should have provided some mock examples.

A) Say there is an MA id annotated with 2 MNX ids. In the meantime, one of the MNX ids is removed because it's no longer correct. Does the MA id stay the same?
B) Say a GPR for a reaction is fixed, does the MA id stay the same?

C) Say the "last" reaction is MA-R10000, which gets removed for some good reason. Then, the "last" reaction becomes MA-R09999. After some time, a new reaction is added. How can one ensure the ID will be MA-R10001?

**Reply by haowang-bioinfo** [reply]
Thanks for the examples. They are not only the cases to be resolved after the implementation, but also questions valid for now.

In my view:
A) Yes.
B) Yes, in general. But still need to see how big changes are to be introduced, a new reaction and id may be added for drastic cases.
C) A very good point. But I believe there is already plenty of solutions to this.

**Reply by mihai-sysbio** [reply]
https://doi.org/10.1371/journal.pbio.2001414 has some good recommendations, especially 
```
Lesson 6. Implement a version-management policy
Lesson 7. Do not reassign or delete identifiers
```

**Reply by haowang-bioinfo** [reply]
Nice reference.
